libnavigator/src/test/java/com/mapbox/navigation/navigator/SensorMapperTest.kt

```diff
@@ -0,0 +1,93 @@
+package com.mapbox.navigation.navigator
+
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import com.mapbox.navigator.SensorType
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class SensorMapperTest {
+
+    @Test
+    fun `should have supported SensorTypesr`() {
+        val supportedSensors = SensorMapper.getSupportedSensorTypes()
+        assertTrue(supportedSensors.isNotEmpty())
+    }
+
+    @Test
+    fun `should map SensorEvent to NavigatorSensorData`() {
+        val sensor: Sensor = mockk(relaxed = true)
+        every { sensor.type } returns Sensor.TYPE_ACCELEROMETER
+        val sensorEvent = mockSensorEvent(
+            sensor,
+            23834291518140L,
+            floatArrayOf(1.151f, 0.164f, 9.700f)
+        )
+
+        val navigationSensorData = SensorMapper.toNavigatorSensorData(sensorEvent)
+        assertNotNull(navigationSensorData!!)
```

Comment:
Open for discussion.

Generally is preferable having only 1 assertion per test 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=57

The problem of having multiple assertions are:
- It's annoying to have to look at the stacktrace to know what's failing (which assertion failed?)
- You don't have information about the remaining assertions (because when an assertion fails it breaks the execution of the test)

That's why is preferable to have only 1 assertion per test and one test per scenario. Basically, if one of them fails you'll know at a glance what in the code might have caused that failure and in which part (class, method, etc.).

I rarely write tests with multiple assertions (at least I try 😅), only if they're tearing apart the result object to verify everything e.g. if there's no `equals` (which is another problem although could happen if it's coming from a 3rd party library object) or if the assertions are atomic to the object. Is this one of these cases?

The same thing applies to the rest of the tests 👇

Reply:
https://github.com/mapbox/mapbox-navigation-android/blob/c325d9d2531a008f87cc14945edf69c3daf2f602/libnavigator/src/test/java/com/mapbox/navigation/navigator/LocationExTest.kt#L27

The goal was to show expected usage of the class, there's no logic being tested so one assert per value might be overkill. Also copied from other tests in the module. 

I would agree with your argument for many tests, but this is testing data mapping. E.g., There's no reason to split a latitude, longitude data mapping into separate tests with 1 assertion each

Reply:
Yeah, if what we're asserting needs to be verified as whole makes total sense to me 🚀 

> if the assertions are atomic to the object

Thanks for clarifying @kmadsen 👍